src/types.rs

```diff
@@ -1555,15 +1861,52 @@ impl Rule {
         });
         lhs.chain(rhs).collect()
     }
-    /// Get a specific subterm in a Rule
+    /// Get a specific subterm in a `Rule`.
+    ///
+    /// See [`Term`] for more information.
+    ///
+    /// [`Term`]: enum.Term.html
+    ///
+    /// # Examples
+    /// 
+    /// ```
+    /// # use term_rewriting::{Signature, Term, Rule, parse_term, parse_rule};
+    /// let mut sig = Signature::default();
+    /// 
+    /// let r = parse_rule(&mut sig, "A(x_) = B | C(x_)").expect("parse of A(x_) = B | C(x_)");
+    ///
+    /// assert_eq!(r.at(&[0]).unwrap().display(&sig), "A(x_)");
+    /// assert_eq!(r.at(&[0,0]).unwrap().display(&sig), "x_");
+    /// assert_eq!(r.at(&[1,0]).unwrap().display(&sig), "B");
+    /// assert_eq!(r.at(&[1,1]).unwrap().display(&sig), "C(x_)");
+    /// ```
```

Comment:
Agh! Your example shouldn't work, but it does because there's a bug in how we implemented `Rule::at`. Could you please change `self.rhs[p[1]].at(&p[2..].to_vec())` to `self.rhs[p[0]].at(&p[1..].to_vec())` and update the examples accordingly?